mreg/api/v1/views_labels.py

```diff
@@ -0,0 +1,62 @@
+from django.core.exceptions import ValidationError
+from django.db.models import Prefetch
+
+from rest_framework import status
+from rest_framework.response import Response
+
+from url_filter.filtersets import ModelFilterSet
+from .views import MregListCreateAPIView, MregRetrieveUpdateDestroyAPIView
+from mreg.models import Label
+from mreg.api.permissions import IsSuperOrAdminOrReadOnly
+from . import serializers
+
+class LabelFilterSet(ModelFilterSet):
+    class Meta:
+        model = Label
+
+
+class LabelList(MregListCreateAPIView):
+    queryset = Label.objects.all()
+    serializer_class = serializers.LabelSerializer
+    permission_classes = (IsSuperOrAdminOrReadOnly,)
+
+    def get_queryset(self):
+        qs = super().get_queryset()
+        return LabelFilterSet(data=self.request.GET, queryset=qs).filter()
+
+    def post(self, request, *args, **kwargs):
+        if "name" in request.data:
+            if self.get_queryset().filter(name=request.data['name']).exists():
+                content = {'ERROR': 'Label name already in use'}
+                return Response(content, status=status.HTTP_409_CONFLICT)
+            if ' ' in request.data['name']:
+                content = {'ERROR': "Label name can't contain spaces"}
+                return Response(content, status=status.HTTP_400_BAD_REQUEST)
+            if not "description" in request.data or request.data['description'] == '':
+                content = {'ERROR': "A description is required for a label"}
+                return Response(content, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
The last two should be handled by the model. Otherwise one can use `PATCH/UPDATE`  to rename to invalid names or descriptions.

Reply:
Removed the last two checks:
https://github.com/unioslo/mreg/blob/17eb02178d5f4025b27754ab3ce4def75e06fd8e/mreg/api/v1/views_labels.py#L27-L33